### PR TITLE
[JENKINS-37575] Keep track of how much content has been copied by writeLog even if the callable is interrupted

### PR DESCRIPTION
**Pull request by jglick**
See [JENKINS-37575](https://issues.jenkins-ci.org/browse/JENKINS-37575) where there is information to reproduce. This is dead code as of #60 + https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/63 but may as well fix the bug so long as the old code path can be run at least with a kill switch.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java [hunk 1]
```diff
@@ -125,43 +125,40 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
...
                 if (len > lastLocation) {
-                    RandomAccessFile raf = new RandomAccessFile(f, "r");
-                    try {
+                    try (RandomAccessFile raf = new RandomAccessFile(f, "r")) {
                         raf.seek(lastLocation);
                         long toRead = len - lastLocation;
                         if (toRead > Integer.MAX_VALUE) { // >2Gb of output at once is unlikely
                             throw new IOException("large reads not yet implemented");
                         }
-                        // TODO is this efficient for large amounts of output? Would it be better to stream data, or return a byte[] from the callable?
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java]: Why is this line removed?

File: src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java [hunk 2]
```diff
@@ -125,43 +125,40 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
...
                         byte[] buf = new byte[(int) toRead];
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java]: Still not a fan of the fact that we chose to do a complete buffer read (since potentially we could have a very large block of data).  I'm itching to change this once these logging PRs are all integrated.

File: src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java (hunk 1 above)
Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java]: Because the speculation is obsolete—the PR is picking another approach entirely.

File: src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java (hunk 2 above)
Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java]: #60 (when active) renders this dead code, replaced by something that (potentially) streams content.

**Comment by jglick** [comment]
`writeLog` counts bytes. If you think there is a bug in #61, ask it there.

**Comment by recampbell** [comment]
@jglick  What is the next step for this PR?

**Comment by jglick** [comment]
@recampbell I am awaiting reviews.

**Comment by svanoort** [comment]
I needed to double check yet another interaction with #61 to make sure this wouldn't need amendment with the changes there but I think we're fine now.
